test-alerts.cpp

```diff
@@ -1,8 +1,56 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
 
 #include "test/catch.hpp"
+
 #include "typewise-alert.h"
+#include "typewise_messagehandler.h"
+
 
-TEST_CASE("infers the breach according to limits") {
+TEST_CASE("infers the breach according to temperature lower than limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
 }
+
+TEST_CASE("infers the breach according to temperature higher than limits") {
+  REQUIRE(inferBreach(32, 20, 30) == TOO_HIGH);
+}
+
+TEST_CASE("No breach,Temperature within limits") {
+  REQUIRE(inferBreach(22, 20, 30) == NORMAL);
+}
+
+TEST_CASE("Checks Temperarture breach based on Passive cooling type") {
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING,40) == TOO_HIGH);
+}
+
+TEST_CASE("Checks Temperarture breach based on HI-Active cooling type") {
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING,-1) == TOO_LOW);
+}
+
+TEST_CASE("Checks Temperarture breach based on Med-Active cooling type") {
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING,20) == NORMAL);
+}
+
+TEST_CASE("Check and alert Target Controller for MED_ACTIVE_COOLING type with too low temperature ") {
+     BatteryCharacter batterychar= {MED_ACTIVE_COOLING, "BMS Temperature"};
+    REQUIRE(checkAndAlert(TO_CONTROLLER, batterychar,-1) == StatusToController);
```

Comment:
This REQUIRE statement conveys that the job of `checkAndAlert` is to convert the parameters to one return value.
If you had to re-state the real job of checkAndAlert in terms of data-conversion, what would you state?